internal/cmd/operator-sdk/bundle/validate/cmd.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/bundle/validate/internal"
+	"github.com/operator-framework/operator-sdk/internal/flags"
+)
+
+const (
+	longHelp = `The 'operator-sdk bundle validate' command can validate both content and format of an operator bundle
+image or an operator bundle directory on-disk containing operator metadata and manifests. This command will exit
+with an exit code of 1 if any validation errors arise, and 0 if only warnings arise or all validators pass.
+
+More information about operator bundles and metadata:
+https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+
+NOTE: if validating an image, the image must exist in a remote registry, not just locally.
+`
+
+	examples = `The following command flow will generate test-operator bundle manifests and metadata,
+then validate them for 'test-operator' version v0.1.0:
+
+  # Generate manifests and metadata locally.
+  $ make bundle
+
+  # Validate the directory containing manifests and metadata.
+  $ operator-sdk bundle validate ./bundle
+
+To build and validate an image built with the above manifests and metadata:
+
+  # Create a registry namespace or use an existing one.
+  $ export NAMESPACE=<your registry namespace>
+
+  # Build and push the image using the docker CLI.
+  $ docker build -f bundle.Dockerfile -t quay.io/$NAMESPACE/test-operator:v0.1.0 .
+  $ docker push quay.io/$NAMESPACE/test-operator:v0.1.0
+
+  # Ensure the image with modified metadata and Dockerfile is valid.
+  $ operator-sdk bundle validate quay.io/$NAMESPACE/test-operator:v0.1.0
```

Comment:
```suggestion
  # Use the built-in Makefile targets to build and push your operator. Make sure to define IMG when you call make:
  $ make docker-build docker-push IMG=<some-registry>/<project-name>:<tag>

  **NOTE**: To allow the cluster pull the image the repository needs to be set as public or you must configure an image pull secret.
  
  # Ensure the image with modified metadata and Dockerfile is valid.
  $ operator-sdk bundle validate <some-registry>/<project-name>:<tag>
```

I do not think that we should adopt the nomenclature Namespace to define a registry. it shows lead to misunderstanding.  Also, I do not think that we should add the export instead we should do as it is done in kb in our docs and other places here now. 

> `mycommand  IMG=<some-registry>/<project-name>:<tag>`


Also, IMO we should tell for the users use the make targets which reduces the steps and make its easier.

Reply:
Also, leI think we need to try to keep a consistency with the doc as well. See; https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/

Reply:
The image tag format goes: `<registry host>/<registry namespace>/<repository name>:<tag>`. I don't see how using different names for something well understood helps users. The docs clearly state "registry namespace" too, so its pretty clear what this refers to.

Will update these docs to use "memcached-operator".

Reply:
Also updated `NAMESPACE -> REG_NAMESPACE`.

Reply:
IHMO we ought add here the same quickstart steps. 
If in each place we create a new nomenclature it might start to be confusing. 
Also, in POV `NAMESPACE` for your registry can lead to misunderstanding.